landlab/components/landslides/landslide_probability.py

```diff
@@ -0,0 +1,535 @@
+
+Relative wetness and factor-of-safety are based on the infinite slope
+stability model driven by topographic and soils inputs and recharge provided
+by user in a "landslide_driver" file. In addition, the component simulates
+the probability of failure for each node based on Monte Carlo simulations
+of the factor-of-safety as the number of simulations with factor-of-safety
+<= 1.0 divided by the number of simulations.
+
+Modified to be more generic in the inputs for greater usability as well
+as accomodate functionality with new release of Landlab version 1.
+
+.. codauthor:: R.Strauch, E.Istanbulluoglu, & S.Nudurupati
+University of Washington
+Created on Thu Aug 20, 2015
+Last edit April 04, 2017
+"""
+
+# %% Import Libraries
+from landlab import Component
+from landlab.utils.decorators import use_file_name_or_kwds
+import numpy as np
+from scipy import interpolate
+from statsmodels.distributions.empirical_distribution import ECDF
+import copy
+
+# %% Instantiate Object
+
+
+class LandslideProbability(Component):
+    """
+    Landlab component designed to calculate a probability of failure at
+    each grid node based on the infinite slope stability model
+    stability index (Factor of Safety).
+
+    The driving force for failure is provided by the user in the form of
+    groundwater recharge; 4 options for providing recharge are included.
+    The model uses topographic and soils characteristics provided as input
+    by the user in the landslide_driver.
+
+    A LandslideProbability calcuation function provides the user with the
+    mean soil relative wetness and probabilty of failure at each node.
+
+    Construction::
+        LandslideProbability(grid, number_of_iterations=250,
+        recharge_minimum=5., groundwater__recharge_maximum=120.)
+
+    Parameters
+    ----------
+    grid: RasterModelGrid
+        A grid.
+    number_of_iterations: float, optional
+        Number of iterations to run Monte Carlo.
+    groundwater__recharge_distribution: str, optional
+        single word indicating recharge distribution, either 'uniform',
+        'lognormal', 'lognormal_spatial,' or 'data_driven_spatial' (None)
+    groundwater__recharge_min_value: float, optional
+        minium groundwater recharge for 'uniform' (mm/d)
+    groundwater__recharge_max_value: float, optional
+        maximum groundwater recharge for 'uniform' (mm/d)
+    groundwater__recharge_mean: float, optional
+        mean grounwater recharge for 'lognormal' (mm/d)
+    groundwater__recharge_standard_deviation: float, optional
+        standard deviation of grounwater recharge for 'lognormal' (mm/d)
+    groundwater__recharge_HSD_inputs: list, optional
+        list of 3 dictionaries in order - HSD_dict {Hydroligic Source
+        Domain (HSD) keys: recharge numpy array values}, {node IDs keys:
+        list of HSD_Id values}, HSD_fractions {node IDS keys: list of
+        HSD fractions values} (none)
```

Comment:
@mcflugen : wait wait wait. I thought the style was Standard Name style, with the double underscore perfectly fine? So this would be `groundwater__recharge_hsd_inputs`? Vast numbers of our other names look like this (e.g., https://github.com/landlab/landlab/wiki/Standard-names, and Table S1 in Hobley et al)